Python 3.8.0 (v3.8.0:fa919fdf25, Oct 14 2019, 10:23:27) 
[Clang 6.0 (clang-600.0.57)] on darwin
Type "help", "copyright", "credits" or "license()" for more information.
>>> 
= RESTART: /Users/aoeshagaedmalsobhe/Library/Mobile Documents/com~apple~CloudDocs/Documents/Practice.py
25564
>>> 
= RESTART: /Users/aoeshagaedmalsobhe/Library/Mobile Documents/com~apple~CloudDocs/Documents/Practice.py
25564
12822
>>> 
= RESTART: /Users/aoeshagaedmalsobhe/Library/Mobile Documents/com~apple~CloudDocs/Documents/Practice.py
25564
12822
Traceback (most recent call last):
  File "/Users/aoeshagaedmalsobhe/Library/Mobile Documents/com~apple~CloudDocs/Documents/Practice.py", line 29, in <module>
    nx.draw(DG)
NameError: name 'DG' is not defined
>>> 
= RESTART: /Users/aoeshagaedmalsobhe/Library/Mobile Documents/com~apple~CloudDocs/Documents/Practice.py
25564
12822
>>> 
= RESTART: /Users/aoeshagaedmalsobhe/Library/Mobile Documents/com~apple~CloudDocs/Documents/Practice.py
25564
Traceback (most recent call last):
  File "/Users/aoeshagaedmalsobhe/Library/Mobile Documents/com~apple~CloudDocs/Documents/Practice.py", line 23, in <module>
    print(len(nx.connected_components(G)))
TypeError: object of type 'generator' has no len()
>>> 
= RESTART: /Users/aoeshagaedmalsobhe/Library/Mobile Documents/com~apple~CloudDocs/Documents/Practice.py
25564
<generator object connected_components at 0x118abd4a0>
12822

= RESTART: /Users/aoeshagaedmalsobhe/Library/Mobile Documents/com~apple~CloudDocs/Documents/Practice.py
25564
876
12822

= RESTART: /Users/aoeshagaedmalsobhe/Library/Mobile Documents/com~apple~CloudDocs/Documents/Practice.py
25564
876
12822

= RESTART: /Users/aoeshagaedmalsobhe/Library/Mobile Documents/com~apple~CloudDocs/Documents/Practice.py
25564
876
12822

= RESTART: /Users/aoeshagaedmalsobhe/Library/Mobile Documents/com~apple~CloudDocs/Documents/Practice.py
25564
Traceback (most recent call last):
  File "/Users/aoeshagaedmalsobhe/Library/Mobile Documents/com~apple~CloudDocs/Documents/Practice.py", line 19, in <module>
    G = nx.DGraph()
AttributeError: module 'networkx' has no attribute 'DGraph'
>>> 
= RESTART: /Users/aoeshagaedmalsobhe/Library/Mobile Documents/com~apple~CloudDocs/Documents/Practice.py
25564
Traceback (most recent call last):
  File "/Users/aoeshagaedmalsobhe/Library/Mobile Documents/com~apple~CloudDocs/Documents/Practice.py", line 23, in <module>
    print(len(list(nx.connected_components(G))))
  File "</Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/decorator.py:decorator-gen-174>", line 2, in connected_components
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/networkx/utils/decorators.py", line 80, in _not_implemented_for
    raise nx.NetworkXNotImplemented(msg)
networkx.exception.NetworkXNotImplemented: not implemented for directed type
>>> 
= RESTART: /Users/aoeshagaedmalsobhe/Library/Mobile Documents/com~apple~CloudDocs/Documents/Practice.py
25564
Traceback (most recent call last):
  File "/Users/aoeshagaedmalsobhe/Library/Mobile Documents/com~apple~CloudDocs/Documents/Practice.py", line 23, in <module>
    print(len(list(nx.connected_components(G))))
  File "</Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/decorator.py:decorator-gen-174>", line 2, in connected_components
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/networkx/utils/decorators.py", line 80, in _not_implemented_for
    raise nx.NetworkXNotImplemented(msg)
networkx.exception.NetworkXNotImplemented: not implemented for directed type
>>> 
= RESTART: /Users/aoeshagaedmalsobhe/Library/Mobile Documents/com~apple~CloudDocs/Documents/Practice.py
25564
876
12822

= RESTART: /Users/aoeshagaedmalsobhe/Library/Mobile Documents/com~apple~CloudDocs/Documents/Practice.py
25564
876
Traceback (most recent call last):
  File "/Users/aoeshagaedmalsobhe/Library/Mobile Documents/com~apple~CloudDocs/Documents/Practice.py", line 27, in <module>
    G.remove_edge(node)
NameError: name 'node' is not defined
>>> 
= RESTART: /Users/aoeshagaedmalsobhe/Library/Mobile Documents/com~apple~CloudDocs/Documents/Practice.py
25564
876
Traceback (most recent call last):
  File "/Users/aoeshagaedmalsobhe/Library/Mobile Documents/com~apple~CloudDocs/Documents/Practice.py", line 27, in <module>
    G.remove_edge(edge)
TypeError: remove_edge() missing 1 required positional argument: 'v'
>>> 
= RESTART: /Users/aoeshagaedmalsobhe/Library/Mobile Documents/com~apple~CloudDocs/Documents/Practice.py
25564
876
12822
24874

= RESTART: /Users/aoeshagaedmalsobhe/Library/Mobile Documents/com~apple~CloudDocs/Documents/Practice.py
25564
876
12822
186
24874

= RESTART: /Users/aoeshagaedmalsobhe/Library/Mobile Documents/com~apple~CloudDocs/Documents/Practice.py
25564
876
12462
66
24634
